packages/web/app/views/patients/panes/EncounterInfoPane.jsx

```diff
@@ -97,13 +88,46 @@ const getDiet = ({ diets }) => {
   );
 };
 
-export const getEncounterType = ({ encounterType }) => <TranslatedEnum
-  enumValues={ENCOUNTER_TYPE_LABELS}
-  value={encounterType}
-/>
+export const getEncounterType = ({ encounterType }) => (
+  <TranslatedEnum enumValues={ENCOUNTER_TYPE_LABELS} value={encounterType} />
+);
 
 const referralSourcePath = 'referralSourceId';
 
+const LengthOfStayText = styled.span`
+  font-weight: 400;
+  color: ${props => props.theme.palette.text.darkestText};
+`;
+
+const LengthOfStayDisplay = ({ encounter }) => {
+  const endDate = encounter.endDate ? parseISO(encounter.endDate) : new Date();
+  const startDate = parseISO(encounter.startDate);
+
+  const duration = intervalToDuration({ start: startDate, end: endDate });
+  const totalMinutes = differenceInMinutes(endDate, startDate);
+
+  let formattedDuration;
+  if (totalMinutes === 0) {
+    formattedDuration = formatDuration(duration, { format: ['seconds'] });
+  } else if (totalMinutes < 60) {
+    formattedDuration = formatDuration(duration, { format: ['minutes'] });
+  } else if (totalMinutes < 1440) {
+    formattedDuration = formatDuration(duration, { format: ['hours'] });
+  } else {
+    formattedDuration = formatDuration(duration, { format: ['days'] });
+  }
+
+  if (!formattedDuration) return null;
+
+  return (
+    <LengthOfStayText>
+      (<TranslatedText stringId="encounter.summary.lengthOfStay.label" fallback="LOS" />
+      {' - '}
+      {formattedDuration})
+    </LengthOfStayText>
+  );
+};
+
```

Comment:
```suggestion
const LengthOfStayDisplay = ({ startDate, endDate }) => {
  const endDate = endDate ? parseISO(endDate) : new Date();
  const startDate = parseISO(encounter.startDate);

  const duration = intervalToDuration({ start: startDate, end: endDate });
  const totalMinutes = differenceInMinutes(endDate, startDate);

  const formatMap = [
    { limit: 1, units: ['seconds'] },
    { limit: 60, units: ['minutes'] },
    { limit: 1440, units: ['hours'] },
    { limit: Infinity, units: ['days'] },
  ];

  const { units } = formatMap.find(({ limit }) => totalMinutes < limit);
  const formattedDuration = formatDuration(duration, { format: units });
  
  if (!formattedDuration) return null;

  return (
    <LengthOfStayText>
      (<TranslatedText
        stringId="encounter.summary.lengthOfStay.label"
        fallback="LOS"
      />
      {' - '}
      {formattedDuration})
    </LengthOfStayText>
  );
};
```
Optional suggestion but I reckon you could do something like this to reduce repetition and also I think it's good practise to narrow the input params as much as possible.